src/main/scala/scorex/core/network/PeerConnectionHandler.scala

```diff
@@ -180,7 +181,7 @@ class PeerConnectionHandler(val settings: NetworkSettings,
       chunksBuffer = t._2
 
       t._1.find { packet =>
-        messagesHandler.parseBytes(packet.toByteBuffer, Some(selfPeer.get)) match {   //todo: .get
+        messagesHandler.parseBytes(packet.toByteBuffer, selfPeer) match {
```

Comment:
I suspect that this change breaks the semantics. When `selfPeer == None`, the old code would clearly result in `Failure(e)`, whereas it is not clear what happens in the new code. Will `parseBytes` throw an exception when `selfPeer` is `None`?

Reply:
I actually think that the semantics before the change would raise the exception and not `Failure(e)`. The expression `Some(selfPeer.get)` is a by-value parameter in `parseBytes`
```
def parseBytes(bytes: ByteBuffer, sourceOpt: Option[ConnectedPeer]): Try[Message[_]]
```
and hence the exception should be thtown before calling the method. This exception is not being caught. 
The only use of `sourceOpt` inside `parseBytes` method is actually at the very return value and it is 
```
Message(spec, Left(msgData), sourceOpt)
```
Hence, it is not even checked.
This is still a different semantics than the one before the change, but I am not sure about the original intention. I will make a general comment about this PR to go further on this.